sandbox/libs/authn/docs/openapi.yaml

```diff
@@ -0,0 +1,79 @@
+openapi: 3.0.0
```

Comment:
Both are valid for openapi specification. I used YAML because that was what I am more familiar with but I don't have a strong preference.

Reply:
After I have a discussion with someone from clients team, I'll post an update once i know the format to go with

Reply:
@DarshitChanpura No, we are not using openapi specification. I made a suggestion (https://github.com/opensearch-project/OpenSearch/issues/6127#issuecomment-1413821232) in the issue @scrawfor99 opened for the permission grant API that its helpful to work backwards from API design. By thinking upfront about how users will interact with the API and coming up with different scenarios for interacting with the API, it will guide what tests should be developed. 

@scrawfor99 Thank you for going through the exercise of creating an openapi yaml file! OpenApi isn't currently used for the native rest APIs in OpenSearch, but there are other areas in the OpenSearch ecosystem where it is currently used. An advantage of using specifications like OpenAPI is that it could lead to code generation which provides a lot of scaffolding for development. 

This file is purely documentation and it helps in reviewing the design of the API.

Reply:
Yeah, I know you said it did not need to be in this format but I thought it may be good since you seemed like you were familiar with it and I read it was machine parseable. That way in the future people could implement an extension for automating permissions grants or stuff

Reply:
The exercise is for you. When designing an API it's helpful to work backwards from the API contract before implementation. Since design of the API affects implementation, its useful to review the design first.